lib/puppet/type/file.rb

```diff
@@ -393,6 +393,22 @@ def self.title_patterns
 
     self.warning "Checksum value is ignored unless content or source are specified" if self[:checksum_value] && !self[:content] && !self[:source]
 
+    if Puppet.features.microsoft_windows? && parameter(:source) &&
+      [:use, :use_when_creating].include?(self[:source_permissions])
+      err_msg = "Copying owner/mode/group from the source file on Windows" <<
+      " is not supported; use source_permissions => ignore."
+
+      if self[:owner] == nil || self[:owner] == :absent || self[:group] == nil ||
+        self[:group] == :absent || self[:mode] == nil || self[:mode] == :absent
```

Comment:
Created https://tickets.puppetlabs.com/browse/PUP-5919 to review the failure behavior later. For now I've implemented it to preserve the prior behavior.

Also, I can't find a way that owner/group/mode get set to absent, so I removed those checks.